### PR TITLE
Feat/#41 exception handling

### PR DESCRIPTION
**Pull request by BalloonWen**
related RFC document: https://github.com/networknt/light-rfcs/commit/f440c47e7b2931995ec7263f6496063106d76416
related issue: https://github.com/networknt/light-router/issues/41
**before change**: in client module whenever there's an error, throw exceptions, then ExceptionHandler will catch it and respond to the client (lost detail information)
**after change**: return a com.networknt.monad.Result, the user can handle based on the "Result" is success or failure. "Result" will keep failure information if fails.

### CODE REVIEW

**Comment by BalloonWen** [comment]
now whenever needs to pass auth server's response to the client, now it will escape based on the type that the server returns. @NicholasAzar 
https://github.com/networknt/light-4j/pull/380/commits/4b354196cd5b77cf2315e3343088f6e0d776943c

**Comment by NicholasAzar** [comment]
@BalloonWen 
{"statusCode":401,"code":"ERR10052","message":"GET_TOKEN_ERROR","description":"Cannot get valid token: "{\\r\\n  \\"error\\":\\"invalid_client\\",\\r\\n  \\"error_description\\":\\"The given client credentials were not valid\\"\\r\\n}".","severity":"ERROR"}

Json format error is: 
"Expected ',' instead of '{'"

**Comment by NicholasAzar** [comment]
Looks better! : 
{
    "statusCode": 401,
    "code": "ERR10052",
    "message": "GET_TOKEN_ERROR",
    "description": "Cannot get valid token: {\r\n  \"error\":\"invalid_client\",\r\n  \"error_description\":\"The given client credentials were not valid\"\r\n}.",
    "severity": "ERROR"
}

File: client/src/main/java/com/networknt/client/oauth/Jwt.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
+package com.networknt.client.oauth;
+
+import com.networknt.config.Config;
+
+import java.util.Map;
+
+public class Jwt {
```
Comment (NicholasAzar) [inline client/src/main/java/com/networknt/client/oauth/Jwt.java]: Class level comments

File: http-string/src/main/java/com/networknt/httpstring/ContentType.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.networknt.httpstring;
+
+public enum ContentType {
```
Comment (NicholasAzar) [inline http-string/src/main/java/com/networknt/httpstring/ContentType.java]: Class level comments pretty please

**Pull request merged**